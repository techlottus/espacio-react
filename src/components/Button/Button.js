

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


