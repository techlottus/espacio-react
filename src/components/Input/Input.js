import React, { useEffect } from 'react'

export const Input = React.memo(({data,value,hasError,errorMessage ,eventFocus,eventKeyPress, eventBlurPress, listenIcon})  => {
    const inputRef = React.createRef();

    useEffect(() => {
        inputRef.current.info = {
            label: data.label || '',
            name: data.name || '',
            type: data.type || '',
            typeButton: data.typeButton || '',
            maxlength: data.maxlength || '',
            onPaste: data.onPaste,
            placeholder: data.placeholder || '',
            autocomplete: data.autocomplete || '',
            disabled: data.disabled,
            alphanumeric: data.alphanumeric,
            alphabetical: data.alphabetical,
            onlyNumbers: data.onlyNumbers,
            upperCase: data.upperCase,
            pattern: data.pattern || '',
            mask: data.mask || ""
        }
    },[data]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {
        inputRef.current.value = value;
    },[value]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {
        inputRef.current.hasError = hasError;
        inputRef.current.errorMessage = errorMessage;
    },[hasError,errorMessage]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {

        inputRef.current.addEventListener('eventFocus', eventFocus);
        inputRef.current.addEventListener('eventKeyPress', eventKeyPress);
        inputRef.current.addEventListener('eventBlurPress', eventBlurPress);
        inputRef.current.addEventListener('listenIcon', listenIcon);
    }, []); // eslint-disable-line react-hooks/exhaustive-deps

    return(
        <>
        <lottus-input ref={inputRef}></lottus-input>
        </>
    )
})