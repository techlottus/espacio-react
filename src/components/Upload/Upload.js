import React, { useEffect } from "react";



export const Upload = React.memo(({ data,hasError,errorMessage, onFile,onRemove }) => {
  const uploadRef = React.createRef();

  useEffect(() => {
    uploadRef.current.data = {
      title: data.title || '',
      extensions: data.extensions || '',
      disabled: data.disabled,
      id: data.id,
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    uploadRef.current.hasError = hasError;
    uploadRef.current.errorMessage = errorMessage;
  }, [hasError,errorMessage]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    uploadRef.current.addEventListener("onFile", onFile);
    uploadRef.current.addEventListener("onRemove", onRemove);
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-upload ref={uploadRef}></lottus-upload>
    </>
  );
});
