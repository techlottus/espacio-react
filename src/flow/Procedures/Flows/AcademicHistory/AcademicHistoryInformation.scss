@import "../../../../App.scss";

.academicinfocontainer {
  font-family: "Nunito", sans-serif;
  margin-top: pixeltorem(22px);
  .forminformationacademichistory {
    .academichistoryprofile {
      margin-top: pixeltorem(32px);
    }
    .academichistoryinfonumber {
      margin-top: pixeltorem(32px);

      .academichistoryinfonumbertitle {
        margin-bottom: pixeltorem(16px);
        color: map-get($palette-neutral, n80);
        font-size: pixeltorem(14px);
      }
      .academichistoryitem {
        padding-bottom: pixeltorem(32px);
      }
    }
  }
  .academichistoryupload {
    .academichistoryinfouploadtitle {
      padding-bottom: pixeltorem(36px);
    }
    .academichistoryuploadinput {
      padding-bottom: pixeltorem(52px);
    }
  }
  .academichistorybtns {
    margin-top: pixeltorem(48px);
    margin-bottom: pixeltorem(120px);
    width: 100%;
    .academicbtn {
      height: 100%;
      &:first-child {
        padding-bottom: pixeltorem(10px);
      }
    }
    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      display: flex;
      .academicbtn {
        &:first-child {
          padding: 0 pixeltorem(15px) 0 0;
        }
      }
    }
  }
}
