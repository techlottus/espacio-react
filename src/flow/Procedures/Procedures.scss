@import "../../App.scss";

.procedures-container {
  width: 100%;
  @extend .headerapp-top;
}
