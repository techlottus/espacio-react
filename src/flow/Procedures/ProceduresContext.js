import { createContext } from "react";

export const ProceduresContext= createContext(null) 