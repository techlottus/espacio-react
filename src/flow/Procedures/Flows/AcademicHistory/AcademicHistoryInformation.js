import React, { useContext, useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { CardProfile } from "../../../../components/Card/CardProfile";
import { Input } from "../../../../components/Input/Input";
import {
  cancelBtnAcademicHistory,
  inputPhoneAcademicHistory,
  nextBtnAcademicHistory,
  selectDeliveryCampusAcademicHistory,
  selectTypeDeliveryAcademicHistory,
  textareaAcademicHistory,
} from "../../../../constants/ProceduresAcademicHistory.constant";
import { Select } from "../../../../components/Select/Select";
import { Button } from "../../../../components/Button/Button";
import { ProceduresContext } from "../../ProceduresContext";
import { getAcademicHistoryService } from "../../../../middlewares/proceduresMiddlewares/academicHistoryMiddleware";
import { useHistory } from "react-router";
import {
  typeflowsProcedures,
  typeStagesProcedures,
  typesDelivaryProcedures,
  cardProfileProcedures,
} from "../../../../constants/Procedures.constant";
import { getValuesOfAuth } from "../../../../helpers/auth";
import { useSelectProcedures } from "../../../../hooks/useSelectProcedures";
import { setAcademicHistoryData } from "../../../../actions/proceduresActions/academicHistoryAction";
import { useFormBuilder } from "../../../../hooks/useForm";

import "./AcademicHistoryInformation.scss";
import { resetFlowProcedures } from "../../../../helpers/proceduresActions";
import { getAcademicHistoryValidators } from "../../../../validators/procedures/academicHistoryValidator";
import { typesModality } from "../../../../types/typesProcedures";
import { sendInfoTM } from "../../../../tagging/services/sendInfoTagManager";
import { cancelStepOneAcademicHistoryTag, nextStepOneAcademicHistoryTag } from "../../../../tagging/flows/procedures/academicHistoryTag";
import { getImageOfAssets } from "../../../../helpers/getImages";

const AcademicHistoryInformation = () => {
  const { handleSteps } = useContext(ProceduresContext);
  const { texts } = useSelector((state) => state.texts);

  const history = useHistory();
  const dispatch = useDispatch();

  const [valuesForm, setValuesForm] = useState({
    name: null,
    value: null,
  });

  const { information: infoStore, data: dataStore } = useSelector(
    (state) => state.proceduresAcademicHistory
  );

  const { profile: profileProcedure } = useSelector(
    (state) => state.procedures
  );

  const [profile, setProfile] = useState({ 
    ...cardProfileProcedures, 
    image: getImageOfAssets(texts?.accountStatus?.images?.user)
  });

  const [optionsCampusDelivery,setOptionsCampusDelivery] = useSelectProcedures(
    infoStore.optionsCampusDelivery,
    dataStore.selectCampusDelivery
  );
  const [optionsTypeDelivery] = useSelectProcedures(
    infoStore.optionsTypeDelivery,
    dataStore.selectTypeDelivery
  );

  const [valuePhone, setValuePhone] = useState("");

  const [valueComments, setValueComments] = useState("");

  const actionValid = () => {
    dispatch(setAcademicHistoryData(next.value));
    history.push(
      `/procedures-flows/${typeflowsProcedures.academyHistory}/${typeStagesProcedures.documents}`
    );
  };

  const {form, setForm, next, setNext, nextBtn, setNextBtn,errorForms} = useFormBuilder(
    nextBtnAcademicHistory,
    actionValid
  );

  const [isShowCampus, setIsShowCampus] = useState(
    dataStore.selectTypeDelivery === typesDelivaryProcedures.fisico
  );

  const [isShowDelivery] = useState(
    getValuesOfAuth().modality === typesModality.withDelivery
  );

  useEffect(() => {
    if (infoStore.profile !== null) {
      const inforProfile = profileProcedure;
      setProfile((state) => {
        return {
          ...state,
          name: inforProfile?.name,
          items: inforProfile?.items,
        };
      });
    }
    setValuePhone(dataStore.phone || getValuesOfAuth().phone);
    setValueComments(dataStore.comments || "");
  }, [dataStore, infoStore, profileProcedure]);

  useEffect(() => {
    window.scrollTo(0, 0);
    handleSteps(0);

    if (
      dataStore.phone &&
      dataStore.selectTypeDelivery &&
      ((dataStore.selectTypeDelivery === typesDelivaryProcedures.fisico &&
        dataStore.selectCampusDelivery) ||
        dataStore.selectTypeDelivery === typesDelivaryProcedures.digital)
    ) {
      setNextBtn(false);
    } else {
      dispatch(getAcademicHistoryService(history));
    }
    setForm(
      getAcademicHistoryValidators(dataStore, getValuesOfAuth().modality)
    );
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    if (valuesForm.name && valuesForm.value) {
      form.get(valuesForm.name).setValue(valuesForm.value);
      if (valuesForm.name === selectTypeDeliveryAcademicHistory.name) {
        if (valuesForm.value === typesDelivaryProcedures.digital) {
          setOptionsCampusDelivery(infoStore.optionsCampusDelivery, null);
        }
        setForm(
          getAcademicHistoryValidators(
            form.value,
            getValuesOfAuth().modality,
            valuesForm.value
          )
        );
      }
      setValuesForm({
        name: null,
        value: null,
      });
    }
  }, [valuesForm, form]); // eslint-disable-line react-hooks/exhaustive-deps

  const handleIsShowCampus = () => {
    return isShowCampus ? (
      <div className="academichistoryitem">
        <Select
          data={selectDeliveryCampusAcademicHistory}
          options={optionsCampusDelivery}
          onClick={(e) => {
            setValuesForm({
              name: selectDeliveryCampusAcademicHistory.name,
              value: e.detail,
            });
          }}
        />
      </div>
    ) : (
      <></>
    )
  }

  return (
    <>
      <div className="academicinfocontainer">
        {profileProcedure ? (
          <div className="academichistoryprofile">
            <CardProfile data={profile} />
          </div>
        ) : (
          ""
        )}
        <div className="forminformationacademichistory">
          <div className="academichistoryinfonumber">
            <p className="academichistoryinfonumbertitle">
              {texts?.procedures?.phoneText}
            </p>
            <div className="academichistoryitem">
              <Input
                data={inputPhoneAcademicHistory.data}
                value={valuePhone}
                hasError={errorForms[inputPhoneAcademicHistory.data.name]}
                errorMessage={inputPhoneAcademicHistory.errorMessage}
                eventKeyPress={(e) => {
                  setValuesForm({
                    name: inputPhoneAcademicHistory.data.name,
                    value: e.detail.value,
                  });
                }}
              />
            </div>
            {isShowDelivery ? (
              <>
                <div className="academichistoryitem">
                  <Select
                    data={selectTypeDeliveryAcademicHistory.data}
                    options={optionsTypeDelivery}
                    onClick={(e) => {
                      if (e.detail === typesDelivaryProcedures.fisico) {
                        setIsShowCampus(true);
                      } else {
                        setIsShowCampus(false);
                      }

                      setValuesForm({
                        name: selectTypeDeliveryAcademicHistory.name,
                        value: e.detail,
                      });
                    }}
                  />
                </div>
                {handleIsShowCampus()}
              </>
            ) : (
              ""
            )}
            <div className="academichistoryitem">
              <Input
                data={textareaAcademicHistory.data}
                value={valueComments}
                eventKeyPress={(e) => {
                  setValuesForm({
                    name: textareaAcademicHistory.data.name,
                    value: e.detail.value,
                  });
                }}
              />
            </div>
          </div>
        </div>
        <div className="academichistorybtns">
          <div className="academicbtn">
            <Button
              data={{
                ...cancelBtnAcademicHistory,
                isExpand: window.innerWidth < 991,
              }}
              onClick={() => {
                resetFlowProcedures(dispatch);
                history.push(`/procedures-main`);
                sendInfoTM(window, cancelStepOneAcademicHistoryTag)
              }}
            />
          </div>
          <div className="academicbtn">
            <Button
              data={nextBtn}
              onClick={() => {
                sendInfoTM(window, nextStepOneAcademicHistoryTag) 
                setNext(true);
              }}
            />
          </div>
        </div>
      </div>
    </>
  );
};

export default AcademicHistoryInformation;
