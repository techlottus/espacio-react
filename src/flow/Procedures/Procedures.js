import React, { useEffect, useState } from "react";
import HeaderApp from "../../shared/HeaderApp/HeaderApp";
import "./Procedures.scss";
import { ProceduresContext } from "./ProceduresContext";
import {
  stepsProcedures,
  typeflowsProcedures,
} from "../../constants/Procedures.constant";
import ProcedureLayout from "./components/ProcedureLayout/ProcedureLayout";
import { ModalProcedures } from "../../components/Modal/ModalProcedures/ModalProcedures";
import { useHistory } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { proceduresRequest } from "../../middlewares/proceduresMiddlewares/indexProceduresMiddleware";
import { getProceduresCostsService } from "../../middlewares/proceduresMiddlewares/proceduresCostsMiddleware";
import { ModalAccordion } from "../../components/Modal/ModalAccordion/ModalAccordion";
import { getProfileProceduresService } from "../../middlewares/proceduresMiddlewares/procedureMainMiddleware";
import { ModalExit } from "../../components/Modal/ModalExit/ModalExit";
import { setPathExitProcedures } from "../../actions/proceduresActions/proceduresAction";
import { unblockProcedures } from "../../helpers/unblock";
import { sendInfoTM } from "../../tagging/services/sendInfoTagManager";
import {
  modalProcedureBackEquivalenceStudiesTag,
  nextStepTwoEquivalenceStudiesTag,
} from "../../tagging/flows/procedures/equivalenceStudiesTag";
import {
  modalProcedureBackAcademicHistoryTag,
  modalProcedureNextAcademicHistoryTag,
} from "../../tagging/flows/procedures/academicHistoryTag";
import { certificateStudyVerifyTag } from "../../tagging/flows/procedures/certificateStudyTag";
import {
  modalVerifyBackStudyRecordTag,
  modalVerifyNextStudyRecordTag,
} from "../../tagging/flows/procedures/studyRecordTag";
import {
  modalVerifyBackReinstatementTag,
  modalVerifyNextReinstatementTag,
} from "../../tagging/flows/procedures/reinstatementTag";
import { ModalRequirementsProcedure } from "../../components/Modal/ModalRequirementsProcedure/ModalRequirementsProcedure";
import { ModalImage } from "../../components/Modal/ModalImage/ModalImage";

const Procedures = () => {
  const [steps, setSteps] = useState({
    ...stepsProcedures,
  });

  const history = useHistory();
  const [isModal, setIsModal] = useState({
    valid: false,
    info: null,
    file: null,
    type: null,
  });

  const [isModalExit, setIsModalExit] = useState({
    isShow: false,
    unblock: null,
  });

  const [isModalAcordeon, setIsModalAcordeon] = useState(false);
  const [isModalRequirements, setIsModalRequirements] = useState({
    isShow: false,
    info: []
  });
  const [isModalImage, setIsModalImage] = useState(false);
  const [isTabs,setIsTabs] = useState(true);
  const dispatch = useDispatch();
  const { proceduresAllOne,proceduresAllTwo } = useSelector((state) => state.procedures);
  const procedures = useSelector((state) => state.procedures);

  const handleStepsBody = (focus, titles ) => {
    setSteps((state) => {
      return {
        ...state,
        focus,
        titles
      };
    });
  };

  const handleSteps = (focus) => {
    setSteps((state) => {
      return {
        ...state,
        focus,
      };
    });
  };

  const handleAccordion = (state) => {
    setIsModalAcordeon(state);
  };

  const handleIsTabs = (isCheck) => {
    setIsTabs(isCheck);
  }

  const handleModalRequirements = (isShow,info) => {
    setIsModalRequirements({isShow,info});
  };

  const handleModalImage = (state) => {
    setIsModalImage(state);
  };

  const handleModalProcedures = (info) => {
    setIsModal(info);
  };
  const handleRequest = () => {
    setIsModal(false);
    dispatch(
      proceduresRequest(
        isModal.type,
        history,
        isModal.info,
        isModal.file,
        isModal.detailId
      )
    );
  };

  const [infoModal, setInfoModal] = useState({
    profile: null,
    information: null,
    file: null,
    title: null,
    cost: null,
  });
  const handleInfoModalProcedures = (info) => {
    setInfoModal(info);
  };
  useEffect(() => {
    let urlSplit = history.location.pathname.split("/");

    let procedureFlow = urlSplit[2];
    let unblock = history.block((location) => {
      if (unblockProcedures(location.pathname, procedureFlow)) {
        dispatch(setPathExitProcedures(location.pathname));
        setIsModalExit({
          isShow: true,
          unblock,
        });
        return false;
      }

      return true;
    });

    if (proceduresAllOne === null || proceduresAllTwo === null) {
      dispatch(getProfileProceduresService());
      dispatch(getProceduresCostsService());
    }
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <div className="procedures-container">
        <HeaderApp />
        <ProceduresContext.Provider
          value={{
            steps,
            handleStepsBody,
            handleSteps,
            isModal,
            isTabs,
            handleModalProcedures,
            handleInfoModalProcedures,
            handleAccordion,
            handleModalRequirements,
            handleIsTabs,
            handleModalImage
          }}
        >
          <ProcedureLayout />
          <ModalExit
            isShow={isModalExit.isShow}
            onClose={() => {
              setIsModalExit((state) => {
                return {
                  ...state,
                  isShow: false,
                };
              });
            }}
            onNext={() => {
              isModalExit.unblock();
              setIsModalExit({
                isShow: false,
                unblock: null,
              });
              setPathExitProcedures(null);
              history.push(procedures.pathExit);
            }}
            flow={"procedures"}
          />
        </ProceduresContext.Provider>
        <ModalProcedures
          isShow={isModal.valid}
          outside={() => setIsModal(false)}
          onClose={() => {
            setIsModal(false);
            switch (isModal.type) {
              case typeflowsProcedures.certificateStudy:
                sendInfoTM(window, certificateStudyVerifyTag, "back");
                break;
              case typeflowsProcedures.equivalenceStudy:
                sendInfoTM(window, modalProcedureBackEquivalenceStudiesTag);
                break;
              case typeflowsProcedures.academyHistory:
                sendInfoTM(window, modalProcedureBackAcademicHistoryTag);
                break;
              case typeflowsProcedures.studyRecord:
                sendInfoTM(window, modalVerifyBackStudyRecordTag, "back");
                break;
              case typeflowsProcedures.reinstatement:
                sendInfoTM(window, modalVerifyBackReinstatementTag, "back");
                break;
              default:
                break;
            }
          }}
          onNext={() => {
            handleRequest();
            switch (isModal.type) {
              case typeflowsProcedures.certificateStudy:
                sendInfoTM(window, certificateStudyVerifyTag, "next");
                break;
              case typeflowsProcedures.equivalenceStudy:
                sendInfoTM(window, nextStepTwoEquivalenceStudiesTag);
                break;
              case typeflowsProcedures.academyHistory:
                sendInfoTM(window, modalProcedureNextAcademicHistoryTag);
                break;
              case typeflowsProcedures.studyRecord:
                sendInfoTM(window, modalVerifyNextStudyRecordTag, "next");
                break;
              case typeflowsProcedures.reinstatement:
                sendInfoTM(window, modalVerifyNextReinstatementTag, "next");
                break;
              default:
                break;
            }
          }}
          infoModal={infoModal}
        />
        <ModalAccordion
          isShow={isModalAcordeon}
          onClose={() => {
            handleAccordion(false);
          }}
          outside={() => handleAccordion(false)}
        />
        <ModalRequirementsProcedure
          isShow={isModalRequirements.isShow}
          info={isModalRequirements.info}
          onClose={() => setIsModalRequirements({
            isShow: false,
            info: []
          })}
          outside={() => setIsModalRequirements({
            isShow: false,
            info: []
          })}
        />
        <ModalImage
          isShow={isModalImage}
          onClose={() => {
            handleModalImage(false);
          }}
          outside={() => handleModalImage(false)}
        />
      </div>
    </>
  );
};

export default Procedures;
