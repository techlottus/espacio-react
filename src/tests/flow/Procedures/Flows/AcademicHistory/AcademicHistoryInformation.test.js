import React from "react";
import { mount } from "enzyme";
import AcademicHistoryInformation from "../../../../flow/Procedures/Flows/AcademicHistory/AcademicHistoryInformation";
import { Provider } from "react-redux";
import { typesRequestErrors } from "../../../../constants/error.constant";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { ProceduresContext } from "../../../../flow/Procedures/ProceduresContext";
import { BrowserRouter } from "react-router-dom";
import { act } from "@testing-library/react";
import { Button } from "../../../../components/Button/Button";
import {
  fileSuccessMock,
  proceduresAcademicHistoryMock,
} from "../../../../fixtures/academicHistoryDocumentsFixture";
import { Upload } from "../../../../components/Upload/Upload";
import { Input } from "../../../../components/Input/Input";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
window.scrollTo = jest.fn();

let initState = {
  proceduresAcademicHistory: {
    information: {
      optionsCampusDelivery: [],
      optionsTypeDelivery: [],
      profile: null,
      phone: null,
    },
    data: {
      selectCampusDelivery: null,
      selectTypeDelivery: null,
      comments: null,
      phone: null,
    },
    document: null,
    errors: {
      [typesRequestErrors.postAcademicHistory]: {
        isError: false,
        msg: null,
      },
    },
  },
  texts: {
    procedures: {
      phoneText: "666",
    },
  },
};

const setSteps = jest.fn().mockImplementation(() => {});
const setIsModal = jest.fn().mockImplementation(() => {});
const setInfoModal = jest.fn().mockImplementation(() => {});
const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };

const handleMount = (initState) => {
  let store = mockStore(initState);
  return mount(
    <ProceduresContext.Provider value={{ setSteps, setIsModal, setInfoModal }}>
      <Provider store={store}>
        <BrowserRouter>
          <AcademicHistoryInformation history={historyMock} />
        </BrowserRouter>
      </Provider>
    </ProceduresContext.Provider>
  );
};

describe("Test <AcademicHistoryDocuments/>", () => {
  beforeEach(() => {});

  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });

  test("should go to next page", () => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper
        .find(".academichistorybtns")
        .find(Button)
        .last()
        .props()
        .onClick();
    });
    wrapper.update();
  });

  describe("should reset flow", () => {
    test("should reset flow and back to main page", () => {
      const wrapper = handleMount(initState);
      act(() => {
        wrapper
          .find(".academichistorybtns")
          .find(Button)
          .first()
          .props()
          .onClick();
      });
      wrapper.update();
    });
  });

  test('should set values of input', () => {
      const wrapper = handleMount(initState);
      act(() => {
          wrapper.find(".academichistoryitem").find(Input).props().eventKeyPress()
      })
      wrapper.update()
  })
});
