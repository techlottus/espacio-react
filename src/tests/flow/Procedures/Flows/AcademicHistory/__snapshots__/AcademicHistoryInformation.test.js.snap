// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AcademicHistoryDocuments/> should init component 1`] = `
<div
  className="academicinfocontainer"
>
  <div
    className="academichistoryprofile"
  >
    <lottus-card-profile />
  </div>
  <div
    className="forminformationacademichistory"
  >
    <div
      className="academichistoryinfonumber"
    >
      <p
        className="academichistoryinfonumbertitle"
      />
      <div
        className="academichistoryitem"
      >
        <lottus-input />
      </div>
      <div
        className="academichistoryitem"
      >
        <lottus-select />
      </div>
      <div
        className="academichistoryitem"
      >
        <lottus-input />
      </div>
    </div>
  </div>
  <div
    className="academichistorybtns"
  >
    <div
      className="academicbtn"
    >
      <lottus-button />
    </div>
    <div
      className="academicbtn"
    >
      <lottus-button />
    </div>
  </div>
</div>
`;
